src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,186 @@
+                }
+            }
+        }
+    }
+
+    private static void processList(int counter, Task[] list) {
+        if (list[0] != null){
+            printListMessage(counter, list);
+        } else {
+            printListButEmptyMessage();
+        }
+    }
+
+    private static void processDone(String input, Task[] list) {
+        int donePos = input.indexOf("done");
+        if (list[0] != null) {
+            if (input.length() < donePos + 5) {
+                printDoneButNotSpecificMessage(); //when input contains done but no number
+            } else {
+                processValidDone(input, list, donePos); //when input contains done and specified number
+            }
+        } else {
+            printDoneButEmptyMessage(); //when input contains done but list is empty
+        }
+    }
+
+    private static void processValidDone(String input, Task[] list, int donePos) {
+        String itemNumDone = input.substring(donePos + 5, donePos + 6);
+        int itemNum = Integer.parseInt(itemNumDone);
+        list[itemNum - 1].setDone();
+        printDoneMessage(list, itemNum);
+    }
+
+    private static int processTasks(String input, int counter, Task[] list) throws IllegalTaskException{
+        if ( ( input.contains("deadline") || input.contains("event") ) && !input.contains("/")){
+            throw new IllegalTaskException();
+        }
+        if (input.contains("todo")) {
+            String description = input.substring(5);
+            ToDo newTask = new ToDo(description);
+            list[counter] = newTask;
+            counter += 1;
+            printAddedTaskMessage(newTask, counter);
+        } else if (input.contains("deadline")) {
+            int donePos = input.indexOf("/"); //finds pos of '/'
+            String description = input.substring(9,donePos);
+            if (!input.substring(donePos + 1, donePos + 3).equals("by")) {
+                throw new IllegalTaskException();
+            }
+            String date = input.substring(donePos + 4);
+            Deadline newTask = new Deadline(description,date);
+            list[counter] = newTask;
+            counter += 1;
+            printAddedTaskMessage(newTask, counter);
+        } else if (input.contains("event")) {
+            int donePos = input.indexOf("/"); //finds pos of '/'
+            String description = input.substring(6,donePos);
+            if (!input.substring(donePos + 1, donePos + 3).equals("by")) {
+                throw new IllegalTaskException();
+            }
+            String date = input.substring(donePos + 4);
+            Event newTask = new Event(description,date);
+            list[counter] = newTask;
+            counter += 1;
+            printAddedTaskMessage(newTask, counter);
+        } else {
+            System.out.println("Please specify tasks: todo, deadline or event");
+            System.out.println("Example - type in the following: todo read book");
+        }
+        return counter;
+    }
```

Comment:
Consider creating TaskManger class to handle task manipulation as well as storing their data for your task list